app/lib/DBInterface.scala

```diff
@@ -0,0 +1,215 @@
+package lib
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+import cats.implicits._
+import cats.data.EitherT
+
+
+import play.modules.reactivemongo.ReactiveMongoApi
+import reactivemongo.bson._
+import reactivemongo.api.collections.bson.BSONCollection
+import reactivemongo.api.Cursor
+import reactivemongo.api.commands._
+
+import com.github.nscala_time.time.Imports._
+
+import lib.Data._
+
+
+class DBInterface(api: ReactiveMongoApi) {
+
+  // Getting data from the database----------------------------------------------------------------
+  def getReservations(screeningId: ScreeningId): EitherT[Future, Error, List[Reservation]] = {
+    val futureQueryResult: Future[List[BSONDocument]] = for {
+      db <- api.database
+      queryDoc = BSONDocument("screening-id" -> screeningId)
+      reservations <- db.collection[BSONCollection]("reservations")
+        .find(queryDoc)
+        .cursor[BSONDocument]()
+        .collect[List](1000, Cursor.FailOnError[List[BSONDocument]]())
```

Comment:
1000 - what is it? Is it some kind of limit of data? What if we have more? How do you think we can handle it?

Reply:
I meant to replace this value with some limit, that I would read from a config file or from an environment variabkle but I forgot about it. The reason there is a limit, is that I haven't found any other way to collect data then this `collect` method.